tests/core.py

```diff
@@ -561,7 +561,10 @@ def test_py_op(templates_dict, ds, **kwargs):
 
     def test_complex_template(self):
         def verify_templated_field(context):
-            self.assertEqual(context['ti'].task.some_templated_field['bar'][1], context['ds'])
+            try:
+                self.assertEqual(context['ti'].task.some_templated_field['bar'][1], context['ds'])
+            except AssertionError as e:
+                raise BaseException(e)
```

Comment:
Thanks for the comment @skudriashev! I think `self.fail` won't work here, since [it internally raises failureException](https://github.com/python/cpython/blob/v3.6.1/Lib/unittest/case.py#L664) and [that's actually AssertionError by default](https://github.com/python/cpython/blob/v3.6.1/Lib/unittest/case.py#L370). So it also will be captured by TaskInstance.run.